### PR TITLE
Drag&drop UI

### PR DESCRIPTION
**Pull request by Gusvarov**

### CODE REVIEW

File: src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div *ngIf="!files.length" class="dropzone" appDragAndDrop (files)="filesDropped($event)">
```
Comment (RomanDeveloperAcc) [inline src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.html:1]: Plaease, if you have more than 1 attribute - write them in separate lines

File: src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+	<div class="text-wrapper">
+		<div class="centered">Drop your image here or <span>browse</span></div>
+	</div>
+</div>
+<div *ngFor="let file of files">
+  <img *ngIf="file" [src]="file.url" appDragAndDrop (files)="filesDropped($event)">
```
Comment (RomanDeveloperAcc) [inline src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.html:7]: Same as line 1

File: src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  selector: 'app-drag-and-drop',
+  templateUrl: './drag-and-drop.component.html',
+  styleUrls: ['./drag-and-drop.component.scss']
+})
+export class DragAndDropComponent implements OnInit {
+
+  constructor() { }
+
+  
+
```
Comment (RomanDeveloperAcc) [inline src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.ts:14]: Please delete 2 empty lines

File: src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.html (hunk 1 above)
Comment (Gusvarov) [reply src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.html]: fixed

File: src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.ts (hunk 3 above)
Comment (Gusvarov) [reply src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.ts]: fixed

File: src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.html (hunk 2 above)
Comment (Gusvarov) [reply src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.html]: fixed

File: src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.scss [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+.centered {
+    font-size: 1.3em;
+    font-size: 18px;
+    text-align: center;
+    width: 200px;
+    font-family: 'Open Sans';
+}
+
+.centered span {
```
Comment (nabermi2020) [inline src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.scss:24]: Please use sass

File: src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  templateUrl: './drag-and-drop.component.html',
+  styleUrls: ['./drag-and-drop.component.scss']
+})
+export class DragAndDropComponent implements OnInit {
+
+  constructor() { }
+
+  files: FileHandle[] = [];
+
+  filesDropped(files: FileHandle[]): void {
```
Comment (nabermi2020) [inline src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.ts:15]: Please add access modifier

File: src/app/directives/drag-and-drop.directive.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import {
+  Directive,
+  HostBinding,
+  HostListener,
+  Output,
+  EventEmitter
+} from "@angular/core";
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+
+export interface FileHandle {
```
Comment (nabermi2020) [inline src/app/directives/drag-and-drop.directive.ts:10]: Please move this interface to the appropriate folder

File: src/app/directives/drag-and-drop.directive.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  @HostListener('drop', ['$event']) public onDrop(evt: DragEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+    
+    let files: FileHandle[] = [];
+    for (let i = 0; i < evt.dataTransfer.files.length; i++) {
+      const file = evt.dataTransfer.files[i];
+      const url = this.sanitizer.bypassSecurityTrustUrl(window.URL.createObjectURL(file));
+      files.push({ file, url });
+    }
```
Comment (nabermi2020) [inline src/app/directives/drag-and-drop.directive.ts:45]: Add an empty line (separating logical blocks)

File: src/app/directives/drag-and-drop.directive.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    evt.preventDefault();
+    evt.stopPropagation();
+  }
+
+  @HostListener('drop', ['$event']) public onDrop(evt: DragEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+    
+    let files: FileHandle[] = [];
+    for (let i = 0; i < evt.dataTransfer.files.length; i++) {
```
Comment (nabermi2020) [inline src/app/directives/drag-and-drop.directive.ts:41]: Please rewrite it using 'forEach'

File: src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.scss (hunk 4 above)
Comment (Gusvarov) [reply src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.scss]: fixed

File: src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.ts (hunk 5 above)
Comment (Gusvarov) [reply src/app/component/eco-news/create-news/drag-and-drop/drag-and-drop.component.ts]: fixed

File: src/app/directives/drag-and-drop.directive.ts (hunk 6 above)
Comment (Gusvarov) [reply src/app/directives/drag-and-drop.directive.ts]: fixed

File: src/app/directives/drag-and-drop.directive.ts (hunk 7 above)
Comment (Gusvarov) [reply src/app/directives/drag-and-drop.directive.ts:41]: fixed

File: src/app/directives/drag-and-drop.directive.ts (hunk 8 above)
Comment (Gusvarov) [reply src/app/directives/drag-and-drop.directive.ts:37]: It is better to use for loop than forEach for FileList
source: https://stackoverflow.com/questions/50604272/typescript-property-foreach-does-not-exist-on-type-filelist

**Pull request merged**